invisible_cities/core/core_functions_test.py

```diff
@@ -72,6 +80,32 @@ def test_in_range_right_shape(data):
     assert core.in_range(data, -1., 1.).shape == data.shape
 
 
+@given(sorted_unique_arrays)
+def test_in_range_left_open_interval(data):
+    output =core.in_range(data, minval=data[0], left="open")
+    assert not output[0]
+    assert all(output[1:])
+
+
+@given(sorted_unique_arrays)
+def test_in_range_right_open_interval(data):
+    output = core.in_range(data, maxval=data[-1], right="open")
+    assert not output[-1]
+    assert all(output[:-1])
+
+
+@given(sorted_unique_arrays)
+def test_in_range_left_close_interval(data):
+    output =core.in_range(data, minval=data[0], left="close")
+    assert all(output)
+
+
+@given(sorted_unique_arrays)
+def test_in_range_right_close_interval(data):
+    output = core.in_range(data, maxval=data[-1], right="close")
+    assert all(output)
+
```

Comment:
This tests the same result as above despite being a different condition which is a bit confusing. I suppose it's because of the default -np.inf for the lower limit but it seems counter intuitive.

Reply:
Yes, maybe I should specify the default value in the test for clarity.